api/api.proto

```diff
@@ -3229,6 +3268,174 @@ message SdkNodeInspectRequest {
+message NodeDrainJob {
+  // ID of the node drain job
+  string id = 1;
+  // NodeID of the node for which this drain job is running
+  string node_id = 2;
+  // State of the current job
+  NodeDrainJobState state = 3;
+  // Parameters is the original request params for this node drain operation
+  SdkNodeDrainRequest parameters = 4;
+  // CreateTime is the time the job was created
+  google.protobuf.Timestamp create_time = 5;
+  // LastUpdateTime is the time the job was updated
+  google.protobuf.Timestamp last_update_time = 6;
+}
+
+// Defines a node drain response
+message SdkNodeDrainResponse {
+    // ID of the node drain job
+    string id = 1;
+    // Job for this node drain operation
+    NodeDrainJob job = 2 ;
+}
+
+// Defines a request to list all the node drain jobs
+message SdkEnumerateNodeDrainJobsRequest  {
+}
+
+// Defines a response will all the known node drain jobs
+message SdkEnumerateNodeDrainJobsResponse {
+  // Jobs is the list of node drain jobs in the response
+  repeated NodeDrainJob jobs = 1;
+}
+
+// Defines a request to update an existing node drain job
+message SdkUpdateNodeDrainJobRequest {
+  // ID of the rebalance job
+  string id = 1;
+  // State is the new task state to update the job to
+  NodeDrainJobState state = 2;
+}
+
+// Defines the response for an update to an existing job
+message SdkUpdateNodeDrainJobResponse {
+}
+
+// Defines a request to get the status of an existing node drain job
+message SdkGetNodeDrainJobStatusRequest {
+  // ID of the node drain job
+  string id = 1;
+}
+
+// Defines the status of an existing node drain job
+message SdkGetNodeDrainJobStatusResponse {
+  // Job for this node drain operation.
+  NodeDrainJob job = 1;
+  // Summary summarizes the node drain job
+  repeated NodeDrainSummary summary = 2;
+}
+
+// NodeDrainSubOperationState state of the node drain operation
+enum NodeDrainSubOperationState  {
+    // This node drain operation was not configured
+    NODE_DRAIN_SUB_OPERATION_NOT_CONFIGURED = 0;
+    // This node drain operation was configured but is pending
+    NODE_DRAIN_SUB_OPERATION_PENDING = 1;
+    // This node drain opeartion is running
+    NODE_DRAIN_SUB_OPERATION_RUNNING = 2;
+    // This node drain operation has completed
+    NODE_DRAIN_SUB_OPERATION_DONE = 3;
+}
+
+// NodeDrainVolumeSummary summary of the drain operation related to volumes
+message NodeDrainVolumeSummary {
+    // State of this operation
+    NodeDrainSubOperationState state = 1;
+
+    // Number of volumes which have been drained
+    uint64 num_volumes_done = 2;
+
+    // Number of volumes which have not been drained yet
+    uint64 num_volumes_pending = 3;
+}
+
+// Provides a summary of  a node drain job
+message NodeDrainSummary {
+    // Attachment summary
+    NodeDrainVolumeSummary attachment_summary = 1;
+    // Volume replica summary
+    NodeDrainVolumeSummary volume_replica_summary = 2;
+    // Kvdb drain summary
+    NodeDrainSubOperationState kvdb_summary = 3;
```

Comment:
Should we have a summary object for kvdb and cloud drive as well for future? Initially we can only have a state in it similar to NodeDrainVolumeSummary.

Reply:
makes sense. I have made those changes.